appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm

```diff
@@ -434,12 +443,37 @@
                          registeredComponentName eventName)
                        #f))))
 
+       ;; For the HandlesEventDispatching interface
+       (define (dispatchMultiEvent eventName :: java.lang.String
```

Comment:
Sorry, missed that comment. In regards to syntax, I changed it because of this: https://github.com/mit-cml/appinventor-sources/pull/347#issuecomment-89458942

Also, yes the multi-generic event returns the instance of the component that was clicked as a callback variable.